DQM/SiStripMonitorCluster/src/SiStripMonitorCluster.cc

```diff
@@ -732,227 +825,232 @@ void SiStripMonitorCluster::analyze(const edm::Event& iEvent, const edm::EventSe
-	  sdetmes.SubDetTotClusterTH1->Fill(sdetmes.totNClusters);
-      if (subdetswitchtotclusprofon)
-	  sdetmes.SubDetTotClusterProf->Fill(trendVar,sdetmes.totNClusters);
-      if (subdetswitchapvcycleprofon)
-	sdetmes.SubDetClusterApvProf->Fill(tbx_corr%70,sdetmes.totNClusters);
-      if (subdetswitchapvcycleth2on)
-	sdetmes.SubDetClusterApvTH2->Fill(tbx_corr%70,sdetmes.totNClusters);
-      if (subdetswitchdbxcycleprofon) {
-	sdetmes.SubDetClusterDBxCycleProf->Fill(dbxincycle,sdetmes.totNClusters);
-      }
-      if (subdetswitchapvcycledbxprof2on)
-	sdetmes.SubDetApvDBxProf2->Fill(tbx_corr%70,dbx,sdetmes.totNClusters);
-    }	
-
-    if ( globalswitchnclusvscycletimeprof2don )
-      {
-	long long tbx_corr = tbx;
-	int the_phase = apv_phase_collection->getPhase("All");
-	
-	if( the_phase == APVCyclePhaseCollection::nopartition ||
-	    the_phase == APVCyclePhaseCollection::multiphase ||
-	    the_phase == APVCyclePhaseCollection::invalid )
-	  the_phase=30;
-	
-	tbx_corr -= the_phase;
-
-	NclusVsCycleTimeProf2D->Fill( tbx_corr%70 , (int)event_history->_orbit , NStripClusters );
-      }
-  }
-}
+        // plot n 2
+        if ( passBPTXfilter_ and passPixelDCSfilter_ and passStripDCSfilter_ ) {
+          if (globalswitchcstripvscpix)  GlobalABXTH1_CSCP->Fill(tbx%3564);
+        }
+        // plot n 2
+
+        for (std::map<std::string, SubDetMEs>::iterator it = SubDetMEsMap.begin();
+        it != SubDetMEsMap.end(); it++) {
+          std::string sdet = it->first;
+          //std::string sdet = sdet_tag.substr(0,sdet_tag.find_first_of("_"));
+          SubDetMEs sdetmes = it->second;
+
+          int the_phase = APVCyclePhaseCollection::invalid;
+          long long tbx_corr = tbx;
+
+
+          if (SubDetPhasePartMap.find(sdet) != SubDetPhasePartMap.end()) the_phase = apv_phase_collection->getPhase(SubDetPhasePartMap[sdet]);
+          if(the_phase==APVCyclePhaseCollection::nopartition ||
+            the_phase==APVCyclePhaseCollection::multiphase ||
+            the_phase==APVCyclePhaseCollection::invalid) {
+              the_phase=30;
+              //std::cout << " subdet " << it->first << " not valid" << " MR " << MultiplicityRegion <<std::endl;
+            }
+            tbx_corr  -= the_phase;
+
+            long long dbxincycle = event_history->deltaBXinCycle(the_phase);
+            //std::cout<<"dbx in cycle: " << dbxincycle << std::endl;
+            if (globalswitchapvcycledbxth2on && !global_histo_filled) {
+              GlobalApvCycleDBxTH2->Fill(tbx_corr%70,dbx);
+              GlobalDBxTH1->Fill(dbx);
+              GlobalDBxCycleTH1->Fill(dbxincycle);
+              global_histo_filled = true;
+            }
+
+
+            // Fill MainDiagonalPosition plots ***************
+            if (cluster_detsetvektor_pix.isValid()){
+              if (ClusterHisto_){
+                if ( passBPTXfilter_ and passStripDCSfilter_ ){
+                  NumberOfStripClus_vs_BX->Fill(tbx%3564,NStripClusters);
```

Comment:
Something odd happening with the file change history here as this is not code implemented by me.